src/chatd.cpp

```diff
@@ -4329,7 +4360,100 @@ void Chat::handleTruncate(const Message& msg, Idx idx)
 
     // if truncate was received for a message not loaded in RAM, we may have more history in DB
     mHasMoreHistoryInDb = at(lownum()).id() != mOldestKnownMsgId;
+    truncateAttachmentHistory();
+}
+
+void Chat::handleRetentionTime()
+{
+    uint32_t retentionTime = getRetentionTime();
+    if (!retentionTime)
+    {
+        // If retentionTime is disabled
+        return;
+    }
+
+    // Get idx of the most recent msg affected by retention time, if any
+    chatd::Idx idx = CHATD_IDX_INVALID;
+    time_t ts = time(nullptr) - retentionTime;
+    CALL_DB(getIdxByRetentionTime, ts, idx);
+    if (idx == CHATD_IDX_INVALID)
+    {
+        // If there are no messages to remove
+        return;
+    }
+
+    Message *msg = findOrNull(idx);
+    if (msg)
+    {
+       // GUI must detach and free any resources associated with erased messages
+       CALL_LISTENER(onRetentionHistoryTruncated, *msg, idx, getMsgStatus(*msg, idx));
+    }
+
+    // Discard current key, if any
+    CALL_CRYPTO(resetSendKey);
+
+    // Clean affected messages in db and RAM
+    CHATID_LOG_DEBUG("Cleaning messages previous to %d seconds", retentionTime);
+    CALL_DB(retentionHistoryTruncate, idx);
+    truncateByRetentionTime(idx);
+
+    // Clean pending rich links
+    removePendingRichLinks(idx);
```

Comment:
You may need to remove pending reactions, in the corresponding branch. I guess this branch will be merged first.

Reply:
I think so, I've noted to make the adjustment in the corresponding branch once this have been merged